lib/omnibus/software.rb

```diff
@@ -280,6 +280,8 @@ def source(val = NULL)
             "be a kind of `Hash', but was `#{val.class.inspect}'")
         end
 
+        val = canonicalize_source(val)
```

Comment:
Hey @jkeiser it looks like my other comment got lost in a commit, but this is still a bit confusing. The _return_ value of the `canonicalize_source` function is a hash, but the function itself also modifies the given hash in place (`Hash#delete` is a destructive operation). So you're modifying the return hash, then setting it to the value of self. Not only is that a weird reflective operation, but it could be very confusing to users. The "best practice" is to either make this a bang method without reassignment

```ruby
canonicalize_source!(val)
```

or rewrite the method to dup the given hash:

```ruby
val = canonicalize_source(val)
# ...
def canonical_source(hash)
  hash = hash.dup
end
```